src/main/java/com/spotify/futures/CompletableFuturesExtra.java

```diff
@@ -55,4 +61,152 @@ private CompletableFuturesExtra() {
     }
     return new CompletableToListenableFutureWrapper<V>(future);
   }
+
+  /**
+   * Returns a new CompletableFuture that is already exceptionally completed with
+   * the given exception.
+   *
+   * @param throwable the exception
+   * @return the exceptionally completed CompletableFuture
+   */
+  public static <T> CompletableFuture<T> exceptionallyCompletedFuture(Throwable throwable) {
+    final CompletableFuture<T> future = new CompletableFuture<T>();
+    future.completeExceptionally(throwable);
+    return future;
+  }
+
+  /**
+   * Returns a new CompletionStage that, when this stage completes
+   * either normally or exceptionally, is executed with this stage's
+   * result and exception as arguments to the supplied function.
+   *
+   * <p>When this stage is complete, the given function is invoked
+   * with the result (or {@code null} if none) and the exception (or
+   * {@code null} if none) of this stage as arguments, and the
+   * function's result is used to complete the returned stage.
+   *
+   * This differs from
+   * {@link java.util.concurrent.CompletionStage#handle(java.util.function.BiFunction)}
+   * in that the function should return a {@link java.util.concurrent.CompletionStage} rather than
+   * the value directly.
+   *
+   * @param fn the function to use to compute the value of the
+   * returned CompletionStage
+   * @param <U> the function's return type
+   * @return the new CompletionStage
+   */
+  public static <T, U> CompletionStage<U> handleCompose(
+          CompletionStage<T> future,
+          final BiFunction<? super T, Throwable, CompletionStage<? extends U>> fn) {
```

Comment:
It is consistent in that it tries to use final in parameters when it's needed for reachability by inner anonymous classes.